boa/contracts/vvm/vvm_contract.py

```diff
@@ -55,6 +71,22 @@ def deploy(self, *args, env=None):
 
         return self.at(address)
 
+    def deploy_as_blueprint(
+        self, env=None, blueprint_preamble=DEFAULT_BLUEPRINT_PREAMBLE, **kwargs
+    ):
+        """
+        Deploy a new blueprint from this contract.
+        :param blueprint_preamble: The preamble to use for the blueprint.
+        :param env: The environment to deploy the blueprint in.
+        :param kwargs: Keyword arguments to pass to the environment `deploy_code` method.
+        :returns: A contract instance.
+        """
+        if env is None:
+            env = Env.get_singleton()
+        bytecode = generate_blueprint_bytecode(self.bytecode)
+        address, _ = env.deploy_code(bytecode=bytecode)
+        return ABIContractFactory.from_abi_dict([]).at(address)
```

Comment:
best to cache the ABIContractFactory, it may create a perf issue

Reply:
I disagree, constructors shouldn't do heavy logic IMO

Reply:
But they do sometimes, and allocation can be unpredictably expensive